### DISCUSSION TITLE
Run Playwright in UI mode

### DISCUSSION THREAD

**Discussion by riddla** [nuxt/test-utils#780] [Q&A]
I'm aware that @danielroe and @mxschmitt are working on something regarding Playwright over at https://github.com/nuxt/test-utils/tree/feat/playwright-runner.

In the meantime I am wondering if one can launch Playwright in UI mode via @nuxt/test-utils? Debugging e2e without a headed browser can be kind of cumbersome. I tried fiddling around with `page.pause` and ` browserOptions.launch.headless = false` but could not get the browser to pause in my test(s).

**Comment by mxschmitt** [comment]
This depends on the work in the linked branch / #776.

**Reply by riddla** [reply]
Thanks for your answer. Now as it seems that the branch is merged/released – do you see a way to launch Playwright in UI mode?

**Reply by mxschmitt** [reply]
I think proper updated Nuxt docs will follow, but for the time being I recommend having something like in [this](https://github.com/nuxt/test-utils/tree/main/examples/app-playwright) example project. Make sure to:

1. Adopt a similar playwright.config.ts as in the example
2. Adopt similar tests like in the example
3. Install `@playwright/test` (no need for playwright-core or playwright package etc.)

**Reply by danielroe** [reply]
See https://nuxt.com/docs/getting-started/testing#testing-with-playwright-test-runner for current docs, and improvements and suggestions very welcome 🙏
